examples/osqp_demo.c

```diff
@@ -60,16 +178,21 @@ int main(void) {
   printf("\n");
 
   /* Setup solver */
-  exitflag = osqp_setup(&solver, P, q, A, l, u, m, n, settings);
+  exitflag = osqp_setup(&solver, qp_P, qp_q, qp_A, qp_l, qp_u, qp_m, qp_n, settings);
 
   /* Solve problem */
   if (!exitflag) exitflag = osqp_solve(solver);
 
   /* Cleanup */
   osqp_cleanup(solver);
-  if (A) free(A);
-  if (P) free(P);
-  if (settings) free(settings);
+
+  if( dynamic_matrices ) {
+    if (qp_A) free(qp_A);
+    if (qp_P) free(qp_P);
+  }
+
+  if (settings && dynamic_settings)
+    free(settings);
```

Comment:
Didn't we have c_free for our on data structures?
(This comment applies to other places in the code where we use free)